script.js

```diff
@@ -1,100 +1,332 @@
+  const unidadesPorSeleccionada = parseFloat(document.getElementById("cantidad-equivalente").value);
+
+  if (nombre && !isNaN(costoUnitario) && !isNaN(cantidad) && unidad && !isNaN(unidadesPorSeleccionada)) {
+    const material = { nombre, costoUnitario, cantidad, unidad, unidadesPorSeleccionada };
+    calculadora.agregarMaterial(material);
+   actualizarTablaMateriales(); 
+    guardarDatosEnStorage();
+
+    //Se limpian los campos
+    formProducto.reset();
+
+     // Se muestra modal dependiendo si se completaron o no los campos
+  const modalMaterial = new bootstrap.Modal(document.getElementById('agregarMaterial'));
+  modalMaterial.show();
+} 
+  }
+);
+
+
+//Evento para editar material
+tablaMateriales.addEventListener("click", (e) => {
+  if (e.target.dataset.tipo === "editar-material") {
+    const index = parseInt(e.target.dataset.index);
+    const m = calculadora.materiales[index];
+
+   // Prellena el formulario con los datos y elimina temporalmente el material para reemplazarlo
+    document.getElementById("nombre-material").value = m.nombre;
+    document.getElementById("costo-material").value = m.costoUnitario;
+    document.getElementById("cantidad").value = m.cantidad;
+    document.getElementById("unidad-material").value = m.unidad;
+    document.getElementById("cantidad-equivalente").value = m.unidadesPorSeleccionada;
+
+    // Eliminar y esperar nuevo submit
+    calculadora.materiales.splice(index, 1);
+    guardarDatosEnStorage();
+    actualizarTablaMateriales();
+  }
+});
+
+
+// Evento para editar gasto fijo
+tablaGastosFijos.addEventListener("click", (e) => {
+// Prellena el formulario y elimina el gasto para reemplazarlo
+  if (e.target.dataset.tipo === "editar-gasto") {
+    const index = parseInt(e.target.dataset.index);
+    const g = calculadora.gastosFijos[index];
+
+    document.getElementById("concepto-fijo").value = g.concepto;
+    document.getElementById("costo-fijo").value = g.costo;
+
+    calculadora.gastosFijos.splice(index, 1);
+    guardarDatosEnStorage();
+    actualizarTablaGastosFijos();
+  }
+});
+
+tablaMateriales.addEventListener("click", (e) => {
+  // se confirma con el usuario antes de eliminar un material de la lista
+// se muestra botones de "Confirmar" y "Cancelar" dentro de la fila
+  if (e.target.dataset.tipo === "material") {
+    const fila = e.target.closest("tr");
+    const index = parseInt(e.target.dataset.index);
+
+    if (fila.querySelector(".btn-confirmar")) return;
+
+    fila.innerHTML = `
+      <td colspan="4" class="text-center text-danger">
+        ¿Seguro que quieres eliminar este material?
+        <button class="btn btn-danger btn-sm btn-confirmar" data-index="${index}">Confirmar</button>
+        <button class="btn btn-secondary btn-sm btn-cancelar">Cancelar</button>
+      </td>
+    `;
+
+    fila.querySelector(".btn-confirmar").addEventListener("click", () => {
+      calculadora.materiales.splice(index, 1);
```

Comment:
Delega la responsabilidad a calculadora:
```suggestion
calculadora.eliminarMaterial(index) 
```